src/Pelagos/Util/DOIutil.php

```diff
@@ -218,6 +218,10 @@ public function updateDOI(
         $input .= 'datacite.publisher:' . $this->escapeSpecialCharacters($publisher) . "\n";
         $input .= "datacite.publicationyear:$publicationYear\n";
 
+        //Adding status to the update
+        $doiMetadata = $this->getDOIMetadata($doi);
```

Comment:
Shouldn't we get the status from dataset instead?

Reply:
If we get the status from the dataset, it is redundant to the publish method and we need to validate the doi status that needs to be set based on the dataset status and restriction, which already exists in the DoiConsumer class